src/PageUtils/Devices/DeviceFactory.php

```diff
@@ -0,0 +1,585 @@
+<?php
+
+namespace HeadlessChromium\PageUtils\Devices;
+
+use HeadlessChromium\Exception\InexistentDevice;
+use HeadlessChromium\PageUtils\Devices\Device;
+use HeadlessChromium\PageUtils\Devices\Resolution;
+use HeadlessChromium\PageUtils\Devices\Viewport;
+
+final class DeviceFactory
+{
+    /**
+     * @var array<string, Device> $devices
+     */
+    private array $devices = [];
+
+    /** 
+     * @return void
+     */
+    public function __construct()
+    {
+        foreach ($this->devices() as $device) {
+            $deviceViewport = $device['viewport'];
+
+            $resolution = new class($deviceViewport['width'], $deviceViewport['height'], $deviceViewport['deviceScaleFactor']) extends Resolution {};
+        
+            $viewport = new class($resolution, $deviceViewport['isMobile'], $deviceViewport['hasTouch'], $deviceViewport['isLandscape']) extends Viewport {};
+        
+            $device = new class($device['name'], $device['user-agent'], $viewport) extends Device {};
+        
+            $this->devices[$device['name']] = $device;
+        }
+    }
+
+    /**
+     * @param string $name
+     *
+     * @return Device
+     */
+    public function getDevice(string $name): Device
+    {
+        if (! \in_array($name, $this->devices)) {
+            throw new InexistentDevice("Device {$name} isn't supported.");
+        }
+
+        return $this->devices[$name];
+    }
+
+    /**
+     * @return array<string, string|array<string, int|bool>>
+     */
+    private function devices(): array
```

Comment:
I think it would be better if I remove this array to a separate PHP file - *DeviceDescriptors.php for example, and then include this one inside the DeviceFactory::device() method.